docarray/array/storage/elastic/backend.py

```diff
@@ -0,0 +1,154 @@
+        da_schema = {
+            "mappings": {
+                "dynamic": "true",
+                "_source": {"enabled": "true"},
+                "properties": {
+                    "embedding": {
+                        "type": "dense_vector",
+                        "dims": elastic_config.n_dim,
+                        "index": "true",
+                        "similarity": elastic_config.distance,
+                    },
+                },
+            }
+        }
+        return da_schema
+
+    def _build_client(self, elastic_config):
+
+        client = Elasticsearch(
+            hosts=self._build_hosts(elastic_config),
+            ca_certs=elastic_config.ca_certs,
+            basic_auth=elastic_config.basic_auth,
+        )
+
+        schema = self._build_schema_from_elastic_config(elastic_config)
+
+        if client.indices.exists(index=elastic_config.index_name):
+            client.indices.delete(index=elastic_config.index_name)
+
+        client.indices.create(
+            index=elastic_config.index_name, mappings=schema['mappings']
+        )
+        client.indices.refresh(index=elastic_config.index_name)
+
+        return client
+
+    def _send_requests(self, request):
+        bulk(self._client, request)
+
+    def _refresh(self):
+        self._client.indices.refresh(index=self._config.index_name)
+
+    def _doc_id_exists(self, doc_id):
+        return self._client.exists(index=self._config.index_name, id=doc_id)
+
+    def __getstate__(self):
+        pass
+
+    def __setstate__(self, state):
+        pass
+
+    def _get_storage_infos(self) -> Dict:
+        return {
+            'Backend': 'ElasticConfig',
+        }
+
+    def _update_offset2ids_meta(self):
+        """Update the offset2ids in elastic"""
+        if self._client.indices.exists(index=self._index_name_offset2id):
+            requests = [
+                {
+                    '_op_type': 'index',
+                    '_id': offset_,  # note offset goes here because it's what we want to get by
+                    '_index': self._index_name_offset2id,
+                    'blob': f'{id_}',
+                }  # id here
+                for id_, offset_ in zip(self._offset2ids.ids, self._offset2ids.offsets)
+            ]
+            r = bulk(self._client, requests)
+
+    def _get_offset2ids_meta(self) -> List:
+        """Return the offset2ids stored in elastic
+
+        :return: a list containing ids
+
+        :raises ValueError: error is raised if index _client is not found or no offsets are found
+        """
+        if not self._client:
+            raise ValueError('Elastic client does not exist')
+
+        n_docs = self._client.count(index=self._index_name_offset2id)["count"]
+        if n_docs != 0:
+            offsets = [x for x in range(n_docs)]
+            resp = self._client.mget(index=self._index_name_offset2id, ids=offsets)
+            ids = [x['_source']['blob'] for x in resp['docs']]
+            return ids
```

Comment:
can't we just store all offset2id in a single row within the index (use a fixed id, for instance '0') ?

Reply:
we could,  but this does the loading with a single mget as well